### DISCUSSION TITLE
Writing std::vector, std::string or complex types

### DISCUSSION THREAD

**Discussion by GDamen** [scikit-hep/uproot5#375] [Ideas]
Hello,
first of all, thanks for the amazing work with uproot!

I'm working on a small portable application to convert .csv to .root files in systems where running the full ROOT framework is not an option (oscilloscopes, etc.). It is my understanding that you are currently implementing in uproot4 the writing to output functionalities. Do you plan to also support writing vectors/arrays, strings, or other complex types to branch? I'm asking because uproot3 does not allow it.

Thanks!
Gab

**Comment by jpivarski** [comment]
Vectors/arrays of numbers yes, strings yes, but more complex types than that no. The coverage will be the same as Uproot 3 (you have to dig into the pull requests to find information about saving vectors/arrays of numbers to ROOT files in Uproot 3, but it's there in a somewhat experimental state).

Note that even though Uproot-writing is my top programming priority, I haven't been able to touch it in a few weeks because of a lot of summer conferences and tutorials. I'll likely get more time by August.
